subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/intra_vc_scheduler.go

```diff
@@ -54,27 +74,27 @@ func (s *defaultIntraVCScheduler) getReservedCellList() map[api.ReservationId]Ch
 	return s.virtualReservedCellList
 }
 
-func (s *defaultIntraVCScheduler) allocateCell(cr CellRequest) *VirtualCell {
-	var fullCellList ChainCellList
-	if cr.ReservationId != "" {
-		fullCellList = s.virtualReservedCellList[cr.ReservationId]
+func (s *defaultIntraVCScheduler) schedule(sr schedulingRequest) map[int32][]CellList {
+	var scheduler *topologyAwareScheduler
+	if sr.reservationId != "" {
+		scheduler = s.schedulersReserved[sr.reservationId]
 	} else {
-		fullCellList = s.virtualNonReservedCellList[cr.Chain]
+		scheduler = s.schedulersNonReserved[sr.chain]
 	}
-
-	vcCellView := getCellViewWithPriority(fullCellList, cr.Priority)
-	c := buddyAlloc(vcCellView, cr.Level)
-	if c != nil {
```

Comment:
Previous cell based intra VC scheduling (assume we can support a fraction of cell allocation inside VC) may better than new algo in some cases, so can we also keep original algo, and let user to choose to use which algo by themselves?

If so, pls make comments to compare new and old algo by pros/cons, suitable scenarios, etc.

Reply:
Will consider in future

---
In reply to: [332453251](https://github.com/microsoft/pai/pull/3684#discussion_r332453251) [](ancestors = 332453251)